utils/funcexp/func_round.cpp

```diff
@@ -734,13 +664,109 @@ string Func_round::getStrVal(Row& row, FunctionParm& parm, bool& isNull, Calpont
 int64_t Func_round::getDatetimeIntVal(Row& row, FunctionParm& parm, bool& isNull,
                                       CalpontSystemCatalog::ColType& op_ct)
 {
-  return parm[0]->data()->getIntVal(row, isNull);
+  int32_t s = parm.size() > 1 ? parm[1]->data()->getIntVal(row, isNull) : 0;
+  if (isNull)
+    return 0;
+  s = (s > 6) ? 6 : s;
+  if (s < 0) 
+  {
+    // int64_t x = parm[0]->data()->getDatetimeIntVal(row, isNull);
```

Comment:
It is good to clean the commented text up.

Reply:
Sure, but I'm still not sure, whether it's necessary to do truncate and round before the decimal point, which is not supported in the InnoDB. If it's not, then I will remove the commented part, which I wrote to support this.